distributed/comm/tcp.py

```diff
@@ -436,9 +436,17 @@ async def connect(self, address, deserialize=True, **connection_args):
         kwargs = self._get_connect_args(**connection_args)
 
         try:
-            stream = await self.client.connect(
-                ip, port, max_buffer_size=MAX_BUFFER_SIZE, **kwargs
-            )
+            # server_hostname option (for SNI) only works with tornado.iostream.IOStream
+            if "server_hostname" in kwargs:
```

Comment:
Is it better to just always go via the start_tls branch?

Reply:
Yes, good point!

Reply:
Actually, hang on.  (And don't merge my latest commit - yet).

tornado.tcpclient.connect() sets server_hostname when called with SSL options, so if we simplify this code path, we've got to explicitly set server_hostname to the host.

Reply:
Ah, and if we *aren't* using TLS, calling start_tls will fail.  

I think the original code as written here is cleanest.  I've just pushed 47d6dd6 that reverts this.